tests_with_others/test_deal.py

```diff
@@ -28,8 +28,7 @@ def yet_yet_another_func() -> bool:
         except Exception as err:
             cause_err = err.__cause__
 
-        self.assertIsNotNone(cause_err)
-        self.assertIsInstance(cause_err, RecursionError)
+        self.assertIsNone(cause_err, "Deal can deal with the recursive contracts.")
 
     def test_inheritance_of_postconditions_incorrect(self) -> None:
```

Comment:
Deal also supports inheritance, you just need to specify it explicitly:
https://deal.readthedocs.io/details/contracts.html#deal-inherit

@mristin thank you for actualizing tests! Do you want to actualize this one as well?

Reply:
Hi @orsinium ,
I am short on time at the moment -- so if you want, please go ahead and make a pull request! Otherwise, I'll be only able to look at it in about two weeks time.

Reply:
P.S. after reading https://deal.readthedocs.io/details/contracts.html#deal-inherit -- you might also want to test that the pre-conditions are weakened, not strengthened.

Reply:
> you might also want to test that the pre-conditions are weakened, not strengthened.

Do you have a relevant test for icontract? I can't right away wrap my head around what contravariance would look like for contracts. Well, from the framework perspective, at least.  Currently, deal considers contracts to be invariant, afaik.

Reply:
(Please apologize the terse answer as I am really short on time.)

Consider [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle). Here's a the relevant test: https://github.com/Parquery/icontract/blob/bb686951e36c743a6c2126e592c680562b07a07d/tests/test_inheritance_precondition.py#L16

For a more detailed explanation, see: https://www.eiffel.org/doc/eiffel/ET-_Inheritance, Section "Inheritance and contracts".